src/WebJobs.Extensions.SendGrid/Bindings/SendGridMessageAsyncCollector.cs

```diff
@@ -33,7 +34,7 @@ public Task AddAsync(SendGridMessage item, CancellationToken cancellationToken =
 
             SendGridHelpers.DefaultMessageProperties(item, _config, _attribute);
 
-            if (item.To == null || item.To.Length == 0)
+            if (item.Personalization?.Count == 0)
```

Comment:
I'm wondering if we want to be in the business of providing error messages like this. I was looking through their error docs, and they'll give you very specific reasons why something doesn't work: https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/errors.html#-Personalizations-Errors. That would prevent us from accidentally short-circuiting something that could work in the future (what if they decide to add server-side 'to' templates or something)?

I'll tweak this to match their rules, but I get wary about trying to keep up with the validation rules of a service we don't own. Thoughts?